File: src/LinksNews/Services/Data/Models/ContentCategory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace LinksNews.Services.Data.Models
{
    [Table("contentCategory")]
    public class ContentCategory
    {
        public long? Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }

        [NotMapped]
        public long? PagesCount { get; set; }

        [NotMapped]
        public long? AuthorsCount { get; set; }

        [NotMapped]
        public long? NewsSourcesCount { get; set; }

        [NotMapped]
        public Boolean Selected { get; set; }
    }
}

File: src/LinksNews/angular2App/app/core/domain/content-category.ts
import * as _ from 'lodash';
import { Page } from "./page/page";

export class ContentCategory {
    id: number;
    name: string;
    description: string;
    selected: boolean;

    pagesCount: number;
    authorsCount: number;
    newsSourcesCount: number;

    pages: Array<Page>;

    clone(): ContentCategory {
        let result: ContentCategory = <ContentCategory>_.assign(new ContentCategory(), this);
        return result;
    }

    static from (category: ContentCategory): ContentCategory {
        let result: ContentCategory = new ContentCategory();
        _.assign(result, category);
        return result;
    }
}